crates/bevy_core_pipeline/src/motion_blur/mod.rs

```diff
@@ -0,0 +1,161 @@
+    ///     than they moved between frames; users may find this distracting.
+    pub shutter_angle: f32,
+    /// The quality of motion blur, corresponding to the number of per-pixel samples taken in each
+    /// direction during blur.
+    ///
+    /// Setting this to `1` results in each pixel being sampled once in the leading direction, once
+    /// in the trailing direction, and once in the middle, for a total of 3 samples (`1 * 2 + 1`).
+    /// Setting this to `3` will result in `3 * 2 + 1 = 7` samples. Setting this to `0` is
+    /// equivalent to disabling motion blur.
+    pub samples: u32,
+    #[cfg(all(feature = "webgl", target_arch = "wasm32"))]
+    // WebGL2 structs must be 16 byte aligned.
+    pub _webgl2_padding: bevy_math::Vec3,
+}
+
+impl Default for MotionBlur {
+    fn default() -> Self {
+        Self {
+            shutter_angle: 0.5,
+            samples: 1,
+            #[cfg(all(feature = "webgl", target_arch = "wasm32"))]
+            _webgl2_padding: bevy_math::Vec3::default(),
+        }
+    }
+}
+
+pub const MOTION_BLUR_SHADER_HANDLE: Handle<Shader> =
+    Handle::weak_from_u128(987457899187986082347921);
+
+/// Adds support for per-object motion blur to the app. See [`MotionBlur`] for details.
+pub struct MotionBlurPlugin;
+impl Plugin for MotionBlurPlugin {
+    fn build(&self, app: &mut App) {
+        load_internal_asset!(
+            app,
+            MOTION_BLUR_SHADER_HANDLE,
+            "motion_blur.wgsl",
+            Shader::from_wgsl
+        );
+        app.add_plugins((
+            ExtractComponentPlugin::<MotionBlur>::default(),
+            UniformComponentPlugin::<MotionBlur>::default(),
+        ));
+
+        let Ok(render_app) = app.get_sub_app_mut(RenderApp) else {
+            return;
+        };
+
+        render_app
+            .init_resource::<SpecializedRenderPipelines<pipeline::MotionBlurPipeline>>()
+            .add_systems(
+                Render,
+                pipeline::prepare_motion_blur_pipelines.in_set(RenderSet::Prepare),
+            );
+
+        render_app
+            .add_render_graph_node::<ViewNodeRunner<node::MotionBlurNode>>(
+                core_3d::graph::NAME,
+                core_3d::graph::node::MOTION_BLUR,
+            )
+            .add_render_graph_edges(
+                core_3d::graph::NAME,
+                &[
+                    core_3d::graph::node::END_MAIN_PASS,
+                    core_3d::graph::node::MOTION_BLUR,
+                    core_3d::graph::node::BLOOM, // we want blurred areas to bloom and tonemap properly.
+                ],
+            );
+    }
+
+    fn finish(&self, app: &mut App) {
+        let Ok(render_app) = app.get_sub_app_mut(RenderApp) else {
+            return;
+        };
+
+        let render_device = render_app.world.resource::<RenderDevice>().clone();
+
+        render_app.insert_resource(pipeline::MotionBlurPipeline::new(&render_device));
```

Comment:
Rather than do this, implement FromWorld for your pipeline and use init_resource()